src/raw/aws/fargate_fetcher.go

```diff
@@ -0,0 +1,274 @@
+package aws
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"net/http"
+	"time"
+
+	docker "github.com/docker/docker/api/types"
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/infra-integrations-sdk/persist"
+
+	"github.com/newrelic/nri-docker/src/biz"
+)
+
+// FargateStats holds a map of Fargate container IDs as key and their Docker metrics
+// as values.
+type FargateStats map[string]*docker.Stats
+
+// FargateFetcher fetches metrics from Fargate endpoints in AWS ECS.
+type FargateFetcher struct {
+	client         *http.Client
+	cpuStore       persist.Storer
+	containerStore persist.Storer
+	logger         log.Logger
+	latestFetch    time.Time
+}
+
+// NewFargateFetcher creates a new FargateFetcher with the given HTTP client.
+func NewFargateFetcher(c *http.Client, l log.Logger) (*FargateFetcher, error) {
+	cpuStore, err := persist.NewFileStore(
+		persist.DefaultPath("fargate_container_cpus"),
+		log.NewStdErr(true),
+		60*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	containerStore, err := persist.NewFileStore(
+		persist.DefaultPath("fargate_container_metadata"),
+		log.NewStdErr(true),
+		60*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return &FargateFetcher{client: c, cpuStore: cpuStore, containerStore: containerStore, logger: l}, nil
+}
+
+// InspectContainer looks up for metadata of a container given a containerID.
+func (e *FargateFetcher) InspectContainer(containerID string) (docker.Container, error) {
+	defer func() {
+		if err := e.containerStore.Save(); err != nil {
+			e.logger.Errorf("persisting previous metrics: %v", err)
+		}
+	}()
+	var taskResponse TaskResponse
+	var containerResponse ContainerResponse
+
+	// Try to load container from the cache store.
+	_, err := e.containerStore.Get(containerID, &containerResponse)
+	if err == nil {
+		e.logger.Debugf("found container %s in cache cpuStore", containerID)
+		return containerResponseToDocker(containerResponse), nil
+	}
+
+	err = e.fetchTaskResponse(&taskResponse)
+	if err != nil {
+		return docker.Container{}, err
```

Comment:
why not return a pointer here, like in some other places?

Reply:
@ardias because this data structure is only used in the functions `func attributes(container types.Container)` and  `func labels(container types.Container) []entry`, both taking a valuei instead of pointer. These functions are also used in other parts of the code and I didn't want to change their signature.

Reply:
if the returned struct is not in a healthy state (therefore an empty value is expected by handlers) I rather returning a possible nil pointer so we can manage it

otherwise things like https://github.com/newrelic/nri-docker/pull/22/files#diff-d5c158e9c48d3c54469db640d9836a9eR63 could panic

don't take what's there as stone written as this integration was developed in a rush (you know beta mindset...)

Reply:
if the returned struct is not in a healthy state (therefore an empty value is expected by handlers) I rather returning a possible nil pointer so we can manage it

otherwise things like https://github.com/newrelic/nri-docker/pull/22/files#diff-d5c158e9c48d3c54469db640d9836a9eR63 could panic

don't take what's there as stone written as this integration was developed in a rush (you know beta mindset...)